crates/starknet-devnet/src/main.rs

```diff
@@ -177,7 +170,15 @@ async fn main() -> Result<(), anyhow::Error> {
     let args = Args::parse();
     let (mut starknet_config, server_config) = args.to_config()?;
 
-    set_and_log_fork_config(&mut starknet_config.fork_config, starknet_config.chain_id).await?;
+    // set forking config if fork url provided
+    set_and_log_fork_config(&mut starknet_config.fork_config).await?;
+
+    // if forking enabled then use chain id from forked network
+    if let Some(chain_id) =
+        get_chain_id_from_forked_network_or_none(&starknet_config.fork_config).await?
+    {
+        starknet_config.chain_id = chain_id;
+    }
```

Comment:
We can do two things which are IMO better:
1. Change `get_chain_id_from_forked_network_or_none` to `get_chain_id_from_forked_network` and make it accept url. That way this calling logic would be:
```rust
    if let Some(url) = &starknet_config.fork_config.url {
        let chain_id = get_chain_id_from_forked_network(url).await?;
        starknet_config.chain_id = chain_id;
    }
```

2. Do this chain id setting inside of `set_and_log_fork_config`. That way we wouldn't have to create the JSON-RPC client one extra time (as we do now).

Reply:
`set_and_log_fork_config` is a method that does enough modifications. I dont see a reason to extend its functionality except reusing json rpc client.

Reply:
[b4cda01](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/549/commits/b4cda01171cf0dfc884e7d94d55baeda646a8fcf)